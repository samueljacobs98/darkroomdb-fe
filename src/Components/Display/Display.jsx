import "./Display.scss";
import Stars from "../Stars/Stars";
import { Link } from "react-router-dom";

const Display = ({ data }) => {
  const getRating = () => {
    if (!data.ratings) return -1;
    const size = data.ratings.length;
    const sum = data.ratings
      .map((rating) => rating.rating)
      .reduce((acc, rating) => acc + rating);
    return Math.round((2 * sum) / size) / 2;
  };

  const getFormats = () => {
    if (!data.formats) return "";
    return data.formats.map((format) => format.format).join("/");
  };

  const getPageData = () => {
    return {
      formats: getFormats(),
      rating: getRating(),
      id: data.filmId,
      name: data.name,
      info: data.info,
      iso: data.iso,
      style: data.style,
    };
  };

  const pageData = getPageData();
  console.log(pageData);

  return (
    <div className="display">
      <header className="display__header">
        <h2>{pageData.name}</h2>
        <div className="display__header-rating">
          {pageData.rating !== -1 && (
            <>
              <Stars rating={pageData.rating} />
              <Link
                className="display__header-rating-link"
                to={`/film/${pageData.id}/rating`}
              >
                Add a rating
              </Link>
            </>
          )}
        </div>
      </header>
      <p className="display__info">{pageData.info}</p>
      <footer className="display__footer">
        <p>{pageData.formats}</p>
        <p>{pageData.style}</p>
      </footer>
    </div>
  );
};

export default Display;
