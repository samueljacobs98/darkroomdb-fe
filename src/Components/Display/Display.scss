@use "../../styles/variables" as *;

.display {
  width: 100%;
  max-width: 41rem;
  padding: 2.5rem;
  margin: 1rem;
  background-color: rgba($colour-primary, 0.2);
  display: flex;
  flex-direction: column;
  gap: 5rem;
  &__header {
    display: flex;
    justify-content: space-between;
  }
  &__info {
    font-size: $fs-secondary;

    @media screen and (min-width: $bp-tablet) {
      font-size: $fs-primary;
    }
  }
  &__footer {
    display: flex;
    justify-content: space-evenly;
  }
}
