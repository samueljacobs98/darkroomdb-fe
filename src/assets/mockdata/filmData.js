const filmData = [
  {
    id: 1,
    name: "Kodak ColourPlus 200",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 2,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 3,
    name: "Ilford HP5+",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "black and white",
  },
  {
    id: 4,
    name: "CineStill 800T",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 5,
    name: "Fomapan 200",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 6,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 7,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 8,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 9,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 10,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 11,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 12,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
];

export default filmData;
