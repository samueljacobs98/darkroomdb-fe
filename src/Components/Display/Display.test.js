import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import filmData from "../../assets/mockdata/filmData";
import Display from "./Display";

it("should match component snapshot", () => {
  const { container } = render(
    <MemoryRouter>
      <Display data={filmData} />;
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
