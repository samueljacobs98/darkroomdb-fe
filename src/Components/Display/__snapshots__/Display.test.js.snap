// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match component snapshot 1`] = `
<div>
  <div
    class="display"
  >
    <header
      class="display__header"
    >
      <h2 />
      <div
        class="display__header-rating"
      >
        <a
          class="display__header-rating-link"
          href="/film/undefined/rating"
        >
          Add a rating
        </a>
      </div>
    </header>
    <p
      class="display__info"
    />
    <footer
      class="display__footer"
    >
      <p />
      <p />
    </footer>
  </div>
  ;
</div>
`;
